spec/features/create_etd_spec.rb

```diff
@@ -0,0 +1,235 @@
+    expect(page.find('#pbDissertationUploaded')['style']).to match(/background-image/)
+
+    # upload supplemental file
+    find('input#cbSupplementalFiles').check
+    filename = 'etd_supplemental.txt'
+    expect(page).not_to have_content(filename)
+    # supplemental files uploaded progress checkbox not visible by default
+    expect(page.find('#pbSupplementalFilesUploaded', visible: false)).not_to be_visible
+    supplemental_upload_input = file_upload_elements[1]
+    supplemental_upload_input.attach_file("spec/fixtures/#{filename}")
+    expect(page).to have_content(filename)
+    # a checked box in the progress section is a background image
+    expect(page.find('#pbSupplementalFilesUploaded')['style']).to match(/background-image/)
+
+    # indicate copyrighted material
+    expect(page.find('#pbPermissionsProvided')['style']).to eq '' # rights not yet selected
+    page.select 'does include', from: 'selectPermissionsOptions'
+
+    # provide copyright permissions letters/files
+    # permission files uploaded progress checkbox not visible by default
+    expect(page.find('#pbPermissionFilesUploaded', visible: false)).not_to be_visible
+    filename = 'etd_permissions.pdf'
+    expect(page).not_to have_content(filename)
+    permissions_upload_input = file_upload_elements[11]
+    permissions_upload_input.attach_file("spec/fixtures/#{filename}")
+    expect(page).to have_content(filename)
+    # a checked box in the progress section is a background image
+    expect(page.find('#pbPermissionsProvided')['style']).to match(/background-image/)
+    expect(page.find('#pbPermissionFilesUploaded')).to be_instance_of Capybara::Node::Element
+
+    # apply licenses
+    expect(page.find('#pbRightsSelected')['style']).to eq '' # rights not applied yet
+    click_link('View Stanford University publication license')
+    page.find('input#cbLicenseStanford').check
+    click_link('Close this window')
+    click_link('View Creative Commons licenses')
+    page.select 'CC Attribution license', from: 'selectCCLicenseOptions'
+    click_link('Close this window')
+
+    # set embargo
+    click_link('Postpone release')
+    page.select '6 months', from: 'selectReleaseDelayOptions'
+    click_link('Close this window')
+
+    expect(page.find('#pbRightsSelected')['style']).to match(/background-image/)
+
+    # "submit etd to registrar"
+    accept_alert do
+      page.find('#submitToRegistrar').click # javascript
+    end
+    # page.find waits for this element to appear
+    expect(page.find('#submissionSuccessful')).to have_content('Submission successful')
+    expect(page.find('#submitToRegistrarDiv > p.progressItemChecked')).to have_content('Submitted')
+
+    # fake reader approval
+    reader_progress_list_el = all('#progressBoxContent > ol > li')[8]
+    expect(reader_progress_list_el).to have_content('Verified by Final Reader - Not done')
+    reader_approved = xml_from_registrar.dup.sub(/<readerapproval>Not Submitted<\/readerapproval>/,
```

Comment:
Can this be extracted to a let or a method?

Reply:
I know this is janky;  I specifically chose variable names to help.  Wouldn't a method be outside the block, so the let statements initializing the constant aren't executed?  I guess I could try doing a let statement at the top of the block, right after the `let(:xml_from_registrar)` -- is that what you meant?

Reply:
yes.